apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx

```diff
@@ -117,6 +123,9 @@ export function FaucetButton({
     faucetWalletBalanceQuery.data !== undefined &&
     faucetWalletBalanceQuery.data.value < toUnits("1", 17);
 
+  const [turnstileLoading, setTurnstileLoading] = useState(true);
+  const [turnstileToken, setTurnstileToken] = useState("");
```

Comment:
would like to avoid these by using a form instead

Reply:
something along these lines: https://docs.page/marsidev/react-turnstile/validating-a-token

Reply:
shadcn form good